lib/inferno/test_runner.rb

```diff
@@ -29,12 +29,21 @@ def session_data_repo
       @session_data_repo ||= Repositories::SessionData.new
     end
 
+    def test_run_current_status
+      # forces db refetch of the test run status in case it has been changed to cancelled
+      test_runs_repo.find(test_run.id).status
```

Comment:
Oh, does `find` call `build_entity`, which then automatically get all the results?  Yeah, that's a ton of extra work.

Should I just call it `current_status_for_test_run`?  I'm iffy on 'current' or not -- its redundant but hints why I am calling this instead of just checking test_run.status that is in there already.

Reply:
I'd probably just call if `get_status`, but I haven't been very systematic in naming repository methods. If you prefer `status_for_test_run` I'm fine with that and would probably leave off `current`.

Reply:
Updated to do this direct search.